### PR TITLE
add files for minor release

### LINKED ISSUE #62: Automate building and uploading releases to PyPI
It would be nice to automatically build the package and upload it to PyPI whenever a release is made. Instructions for setting up a GitHub action to do this can be found here: https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

### PR DESCRIPTION
**Pull request by JaroCamphuijsen**
Update release files for the minor 1.1.0 era5cli release
Also add extra line of documentation about year ranges to cli.py
Also add github actions workflow to build and publishes github releases on test.PyPI and PyPI

addresses part of #54
closes #62

### CODE REVIEW

File: .zenodo.json [hunk 1]
```diff
@@ -35,6 +35,10 @@
         {
             "affiliation": "TU Delft",
             "name": "Hut, Rolf"
+        },
+        {
+            "affiliation": "Swedish Meteorological and Hydrological Institute",
+            "name": "Zimmerman, Klaus"
```
Comment (SarahAlidoost) [inline .zenodo.json:41]:
```suggestion
            "name": "Zimmermann, Klaus"
```

File: CITATION.cff [hunk 2]
```diff
@@ -29,25 +29,29 @@ authors:
...
     family-names: Weel
     given-names: Berend
   - 
-    affiliation: TU Delft"
+    affiliation: "TU Delft"
     family-names: Hut
     given-names: Rolf
+  - 
+    affiliation: "Swedish Meteorological and Hydrological Institute"
+    family-names: Zimmerman
```
Comment (SarahAlidoost) [inline CITATION.cff:45]:
```suggestion
    family-names: Zimmermann
```

File: README.rst [hunk 3]
```diff
@@ -33,6 +33,7 @@ With era5cli you can: ​
 - select multiple variables for several months and years
 - split outputs by years, producing a separate file for every year instead of merging them in one file
 - download multiple files at once
+- download ERA5 preliminary back extension
```
Comment (SarahAlidoost) [inline README.rst:36]:
```suggestion
- download ERA5 back extension (preliminary version)
```

File: era5cli/__version__.py [hunk 4]
```diff
@@ -8,6 +8,6 @@
 
 __author__ = ('Ronald van Haren', 'Jaro Camphuijsen', 'Yifat Dzigan',
               'Niels Drost', 'Fakhereh Alidoost', 'Bouwe Andela',
-              'Jerom Aerts', 'Berend Weel', 'Rolf Hut')
+              'Jerom Aerts', 'Berend Weel', 'Rolf Hut', 'Klaus Zimmerman')
```
Comment (SarahAlidoost) [inline era5cli/__version__.py:11]:
```suggestion
              'Jerom Aerts', 'Berend Weel', 'Rolf Hut', 'Klaus Zimmermann')
```

**Review by SarahAlidoost — APPROVED** [review]
@JaroCamphuijsen nice job, I changed `preliminary back extension` to `back extension (preliminary version)` to be the same as its documentation [here](https://confluence.ecmwf.int/display/CKB/ERA5+back+extension+1950-1978+%28Preliminary+version%29%3A+tropical+cyclones+are+too+intense). Feel free to ignore it. 

And some suggestions, see above.

File: CHANGELOG.rst [hunk 5]
```diff
@@ -1,6 +1,12 @@
 Release Notes
 *************
 
+1.1.0 (2020-12-10)
```
Comment (JaroCamphuijsen) [inline CHANGELOG.rst:4]:
```suggestion
1.1.0 (2020-12-14)
```

File: CITATION.cff [hunk 6]
```diff
@@ -29,25 +29,29 @@ authors:
...
     family-names: Hut
     given-names: Rolf
+  - 
+    affiliation: "Swedish Meteorological and Hydrological Institute"
+    family-names: Zimmermann
+    given-names: Klaus
 
 cff-version: "1.0.3"
-date-released: 2019-07-25
+date-released: 2020-12-10
```
Comment (JaroCamphuijsen) [inline CITATION.cff:49]:
```suggestion
date-released: 2020-12-14
```

**Review by JaroCamphuijsen — COMMENTED** [review]
era5cli release postponed by 4 days

File: .github/workflows/publish-to-pypi.yml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    - name: Publish distribution to Test PyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        password: ${{ secrets.test_pypi_password }}
+        repository_url: http://test.pypi.org/legacy/
+    - name: Publish distribution to PyPI
+      if: startsWIth(github.ref, 'refs/tags')
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        password: ${{ secrets.pypi_password }}
```
Comment (JaroCamphuijsen) [inline .github/workflows/publish-to-pypi.yml:36]:
```suggestion
        password: ${{ secrets.pypi_password }}
        
```

File: pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+[build-system]
+# Minimum requirements for the build system to execute.
+requires = ["setuptools", "wheel"]  # PEP 508 specifications.
```
Comment (JaroCamphuijsen) [inline pyproject.toml:3]:
```suggestion
requires = ["setuptools", "wheel"]  # PEP 508 specifications.

```

**Pull request merged**